src/controllers/features/common/checkSavedApplicationController.ts

```diff
@@ -1,30 +1,60 @@
 import { ErrorService } from "../../../services/errorService";
+import { Resource } from "@companieshouse/api-sdk-node";
+import { ApiErrorResponse } from "@companieshouse/api-sdk-node/dist/services/resource";
+import { TransactionList } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
+import { getSavedApplication } from "../../../services/transactions/transaction_service";
+import { deleteAcspApplication } from "../../../services/acspRegistrationService";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
     const session: Session = req.session as any as Session;
-    const savedApplication = await getSavedApplication(session, res.locals.userId);
+    const savedApplication : Resource<TransactionList> | ApiErrorResponse = await getSavedApplication(session, res.locals.userId);
+    logger.debug("savedApplication: " + savedApplication);
     const lang = selectLang(req.query.lang);
     const locales = getLocalesService();
-    if (savedApplication.status === 404) {
-        // Direct to start a new journey if no application found or application was rejected
-        saveDataInSession(req, "resume_application", false);
+
+    if (savedApplication.httpStatusCode === 200) {
+        logger.debug("saved application exists");
+        const transactionlistResource = savedApplication as Resource<TransactionList>;
+        const transactionList = transactionlistResource.resource;
+        const redirectionUrl = getRedirectionUrl(transactionList, session, res, locales, lang);
+        res.redirect(addLangToUrl(await redirectionUrl, lang));
+    } else if (savedApplication.httpStatusCode === 404) {
+        logger.debug("its a new application");
         res.redirect(addLangToUrl(BASE_URL + TYPE_OF_BUSINESS, lang));
-    } else if (savedApplication.status === 204) {
-        // Application found and can be resumed
-        res.redirect(addLangToUrl(BASE_URL + SAVED_APPLICATION, lang));
-    } else if (savedApplication.status === 403) {
-        // Application found and cannot be resumed
-        res.redirect(addLangToUrl(BASE_URL + CANNOT_SUBMIT_ANOTHER_APPLICATION, lang));
     } else {
-        logger.error(`internal server error: ` + savedApplication.status);
+        logger.error("internal server error: " + savedApplication.httpStatusCode);
         const error = new ErrorService();
         error.renderErrorPage(res, locales, lang, BASE_URL + CHECK_SAVED_APPLICATION);
     }
 };
+
+const getRedirectionUrl = async (transactionList: TransactionList|undefined, session: Session, res: Response, locales:any, lang:string): Promise<string> => {
+    const transaction = transactionList?.items[transactionList?.items.length - 1];
+    var url = "";
+    if (transaction?.status !== "closed") {
+        logger.debug("application is open");
+        url = BASE_URL + SAVED_APPLICATION;
+    } else if (transaction.filings![transaction.id + "-1"]?.status === "rejected") {
```

Comment:
May be use constants instead of strings in the conditional statements.

Reply:
resolved